guardian/post-release-action
# feat: Allow `npm-lockfile-version` to be specified

## What does this change?



The v2 npm lockfile contains the version of a package twice:
  - `.version`
  - `.packages."".version`

This is different from a v1 lockfile, which only includes version once.

If a repository is using a v2 lockfile (such as [CDK](https://github.com/guardian/cdk)), this action will fail, reporting 2 changes were expected but 4 were received.

This change adds another input to the action to allow one specify the version of the npm lockfile being used. We then use this to determine how many changes we expect to see in the lockfile.

Lastly, a little more typesafety, for supported package managers, is added via an enum.

See https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion.

## How to test



See added tests.

The build from https://github.com/guardian/cdk/pull/885 should also go green as it switches to use this branch.

## How can we measure success?



It is possible to use this action in repositories that use v2 of the npm lockfile.
